config.go

```diff
@@ -1097,6 +1097,19 @@ func ValidateConfig(cfg Config, usageMessage string) (*Config, error) {
 			"minbackoff")
 	}
 
+	// Newer versions of lnd added a new sub-config for bolt-specific
+	// parameters. However we want to also allow existing users to use the
+	// value on the top-level config. If the outer config value is set,
+	// then we'll use that. Also we make sure that both values can't be set
+	// at the same time.
+	switch {
+	case cfg.SyncFreelist && cfg.DB.Bolt.NoFreeListSync:
+		return nil, fmt.Errorf("--sync-freelist and " +
+			"--db.bolt.noSyncfreelist cannot both be sent")
+	case cfg.SyncFreelist:
+		cfg.DB.Bolt.NoFreeListSync = !cfg.SyncFreelist
```

Comment:
If we're in this case then `cfg.SyncFreelist` is already `false`, so really we just need an `if` checking the first case.

Reply:
Is it? You mean it's already `true` when we're here? 

What I'm trying to do with this line is "promote" the outer config value to the inner DB config value. As many people today that rely on free list sync for w/e reason, would update and find that it's no longer being synced. So this should let them keep setting the original config value, yet have that be reflected in what's actually checked to open the DB.

Reply:
Ah I see what you mean, the DB value defaults to true (the negation of the outer config value).

Reply:
Modified to only retain a single if statement. Updated the commit message as well.

Reply:
Sorry, what I meant is that `cfg.DB.NoFreeListSync` is already false in the second case (as the first case ensures that).

Reply:
if `SyncFreeList` and `NoFreeListSync` are both true, shouldn't this fail? currently we will silently override `NoFreeListSync`, but that seems counterintuitive since really we should be deprecating `SyncFreeList`

while we're at it, should we at also change the naming to be the same? e.g. `SyncFreeList` and `NoSyncFreeList`?

Reply:
> if SyncFreeList and NoFreeListSync are both true, shouldn't this fail?

That's the behavior we want to support. `NoFreeListSync` defaults to true, but people may have `SyncFreeList` set to true already as the DB scoped one is a new config value. With what's implemented now, we keep the same behavior for those that have `SyncFreeList` set. 

> while we're at it, should we at also change the naming to be the same? e.g. SyncFreeList and NoSyncFreeList?

I considered that but the existing `SyncFreeList` value is already pretty widely used. IMO `NoSyncFreeList` is a better name since that's what `bbolt` accepts directly.

Reply:
> That's the behavior we want to support. NoFreeListSync defaults to true, but people may have SyncFreeList set to true already as the DB scoped one is a new config value. With what's implemented now, we keep the same behavior for those that have SyncFreeList set.

Discussed offline, will remove `No` from arg name and default to false. The latter also allows the behavior to be toggled from the command line, which can't be done if the default of a boolean arg is true.

> IMO NoSyncFreeList is a better name since that's what bbolt accepts directly.

Exactly, that's why I'm suggesting we change it to be `NoSyncFreeList`/`SyncFreeList` (rather than `NoFreeListSync` as it is rn)